mesa/agent.py

```diff
@@ -357,7 +358,69 @@ def random(self) -> Random:
         """
         return self.model.random
 
+    def group_by(self, by: Callable | str, result_type: str = "agentset") -> GroupBy:
+        """
+        Group agents by the specified attribute
+
+        Args:
+            by (Callable, str): used to determine what to group agents by
+
+                                * if ``by`` is a callable, it will be called for each agent and the return is used
+                                  for grouping
+                                * if ``by`` is a str, it should refer to an attribute on the agent and the value
+                                  of this attribute will be used for grouping
+            result_type (str, optional): The datatype for the resulting groups {"agentset", "list"}
```

Comment:
A documentation saying that using list is faster than AgentSet would be informative.